sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -493,6 +564,55 @@ async Task wrapper(CancellationToken token)
             return result;
         }
 
+        /// <summary>
+        /// Attempts to read a legacy checkpoint JSON format and extract an offset and a sequence number
+        /// {"PartitionId":"0","Owner":"681d365b-de1b-4288-9733-76294e17daf0","Token":"2d0c4276-827d-4ca4-a345-729caeca3b82","Epoch":386,"Offset":"8591964920","SequenceNumber":960180}
+        /// </summary>
+        private static void TryReadLegacyCheckpoint(Span<byte> data, out long? offset, out long? sequenceNumber)
+        {
+            offset = null;
+            sequenceNumber = null;
+
+            var jsonReader = new Utf8JsonReader(data);
+            try
+            {
+                if (!jsonReader.Read() || jsonReader.TokenType != JsonTokenType.StartObject) return;
+
+                while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
+                {
+                    switch (jsonReader.GetString())
+                    {
+                        case "Offset":
+                            if (!jsonReader.Read() ||
+                                jsonReader.GetString() is not string offsetString ||
+                                !long.TryParse(offsetString, out long offsetValue))
+                            {
+                                return;
+                            }
+
+                            offset = offsetValue;
+                            break;
+                        case "SequenceNumber":
+                            if (!jsonReader.Read() ||
+                                !jsonReader.TryGetInt64(out long sequenceNumberValue))
+                            {
+                                return;
+                            }
+
+                            sequenceNumber = sequenceNumberValue;
+                            break;
+                        default:
+                            jsonReader.Skip();
+                            break;
+                    }
+                }
+            }
+            catch (JsonException)
+            {
+                // ignore
```

Comment:
nit:  Additional context may be helpful here.   I think what you're saying is "Ignore this because if the data is malformed, it will be treated as if the checkpoint didn't exist."